src/index.ts

```diff
@@ -79,16 +95,27 @@ class MagicNixCacheAction {
       }
     }
 
+    this.idslib.addFact("authenticated_env", !anyMissing);
     if (anyMissing) {
       return;
     }
 
+    if (this.noopMode) {
```

Comment:
Why do we need to check noop mode anywhere that's not the `main` function? I would expect to see `const cacheAction = new .....; if cacheAction.noopMode { return }` and otherwise continue on to doing the other stuff.

Reply:
Good idea. Fixup'd.